src/Graphql/Service.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Butler\Service\Graphql;
+
+use Exception;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Log;
+
+abstract class Service
+{
+    public readonly string $url;
+    public readonly string $token;
+
+    protected int $timeout = 10;
+
+    public function __construct()
+    {
+        $config = config(static::configKey());
+
+        $this->url = $config['url'];
+        $this->token = $config['token'];
+    }
```

Comment:
Wondering if it should be possible to override the default config key as an argument in the constructor but maybe it's enough to add it if the need arises. Or maybe you should just use the `Butler\Graphql\Client` class in that case?

Reply:
Instead of extending `static::configKey()`?

Reply:
Ahh didn't think of that, forget what I said 🙃